tuareg.el

```diff
@@ -2811,6 +2824,12 @@ Short cuts for the Tuareg mode:
 Short cuts for interactions with the REPL:
 \\{tuareg-interactive-mode-map}"
 
+  (setq mode-name
+        '(:eval
+          (let ((other-file (tuareg--other-file (buffer-file-name))))
+            (if other-file
+                (format "Tuareg[+%s]" (file-name-extension other-file))
+              "Tuareg"))))
```

Comment:
Introducing a `tuareg-mode-name` — used as a symbol for `mode-name` — would be neater I think.  We may also want to introduce a variable (`tuareg-name`??) so that the replacement of `Tuareg` with `🐫` still works.